vertx-sql-client/src/main/java/io/vertx/sqlclient/Transaction.java

```diff
@@ -26,7 +26,7 @@
  * A transaction that allows to control the transaction and receive events.
  */
 @VertxGen
-public interface Transaction extends SqlClient {
```

Comment:
Can you clarify is it necessary that `Transaction` does not extend `SqlClient` anymore? I think it becomes more complicated since users have to use the connection API directly if doing some operations on a transaction.

Reply:
I removed it for several reasons:

- I wanted to have the ability to have `Transaction` extends `Future<Void>` instead of having a `result()` field
- we might add in the future the `withTransaction` method to `SqlClient` and in this case it does not make sense to have `withTransaction` on `Transaction` in this situation

Reply:
how about duplicating the two interfaces `query` and `preparedQuery` on `Transaction` so that we can use `Transaction` instread of using the `connection` reference to schedule a command.

Reply:
I believe `Transaction` should remain only there for managing transactions.

Reply:
in other words, having `Transaction` extending `SqlClient` was a design mistake I did and I would like to fix it

Reply:
ok, it seems that we could also remove `Transaction#prepare` in this interface?

Reply:
yes totally, I think I forgot that and I will remove it, thanks for noticing

Reply:
I will also likely remove the `abortHandler` as it is now replaced with the fact that the transaction has a Future<Void> that gives the overral transaction result and thus `onComplete` can be called instead.